File: robocode-tankroyale-server/src/main/java/dev/robocode/tankroyale/server/mappers/BotIntentToBotIntentMapper.java
package dev.robocode.tankroyale.server.mappers;

import dev.robocode.tankroyale.server.model.BotIntent;

public final class BotIntentToBotIntentMapper {

	private BotIntentToBotIntentMapper() {}

	public static BotIntent map(dev.robocode.tankroyale.schema.BotIntent intent) {

		return BotIntent.builder()
			.targetSpeed(intent.getTargetSpeed())
			.turnRate(intent.getTurnRate())
			.gunTurnRate(intent.getGunTurnRate())
			.radarTurnRate(intent.getRadarTurnRate())
			.bulletPower(intent.getFirepower())
			.adjustGunForBodyTurn(intent.getAdjustGunForBodyTurn())
			.adjustRadarForGunTurn(intent.getAdjustRadarForGunTurn())
            .bodyColor(intent.getBodyColor())
            .turretColor(intent.getTurretColor())
            .radarColor(intent.getRadarColor())
            .bulletColor(intent.getBulletColor())
            .scanColor(intent.getScanColor())
			.tracksColor(intent.getTracksColor())
			.gunColor(intent.getGunColor())
			.build();
	}
}

File: server/src/main/kotlin/dev/robocode/tankroyale/server/mapper/BotIntentToBotIntentMapper.kt
package dev.robocode.tankroyale.server.mapper

import dev.robocode.tankroyale.schema.BotIntent

object BotIntentToBotIntentMapper {
    fun map(intent: BotIntent): dev.robocode.tankroyale.server.model.BotIntent {
        intent.apply {
            return dev.robocode.tankroyale.server.model.BotIntent(
                targetSpeed = targetSpeed,
                turnRate = turnRate,
                gunTurnRate = gunTurnRate,
                radarTurnRate = radarTurnRate,
                firepower = firepower,
                adjustGunForBodyTurn = adjustGunForBodyTurn,
                adjustRadarForGunTurn = adjustRadarForGunTurn,
                scan = scan,
                bodyColor = bodyColor,
                turretColor = turretColor,
                radarColor = radarColor,
                bulletColor = bulletColor,
                scanColor = scanColor,
                tracksColor = tracksColor,
                gunColor = gunColor,
            )
        }
    }
}